builtin/providers/aws/resource_aws_instance.go

```diff
@@ -129,9 +129,11 @@ func resourceAwsInstance() *schema.Resource {
 			},
 
 			"network_interface_id": &schema.Schema{
-				Type:     schema.TypeString,
-				Computed: true,
-				Optional: true,
+
+				Type:          schema.TypeString,
```

Comment:
Hi @chiradeep,
Thanks for input.

The problem with the list is that _InstanceNetworkInterfaceSpecification_ requires network interface id **and** index. So Terraform template operator would have to define both at instance creation time and any change would rebuild whole instance (rather undesirable). Additionally it complicates instance resource a bit. That is why I preferred more "atomic" approach, with ability to attach eth0 of given id (versus originally hard-coded new interface) and attaching other enis explicitly (see https://github.com/hashicorp/terraform/pull/10594)